srml/babe/src/lib.rs

```diff
@@ -106,18 +106,77 @@ impl ProvideInherentData for InherentDataProvider {
 	}
 }
 
+/// The length of the BABE randomness
+pub const RANDOMNESS_LENGTH: usize = 32;
+
 decl_storage! {
 	trait Store for Module<T: Trait> as Babe {
 		/// The last timestamp.
 		LastTimestamp get(last): T::Moment;
 
 		/// The current authorities set.
 		Authorities get(authorities): Vec<AuthorityId>;
+
+		/// The epoch randomness.
+		///
+		/// # Security
+		///
+		/// This MUST NOT be used for gambling, as it can be influenced by a
+		/// malicious validator in the short term.  It MAY be used in many
+		/// cryptographic protocols, however, so long as one remembers that this
+		/// (like everything else on-chain) is public.  For example, it can be
+		/// used where a number is needed that cannot have been chosen by an
+		/// adversary, for purposes such as public-coin zero-knowledge proofs.
+		EpochRandomness get(epoch_randomness): [u8; 32];
+
+		/// The randomness under construction
+		UnderConstruction: [u8; 32];
+
+		/// The randomness for the next epoch
+		NextEpochRandomness: [u8; 32];
+
+		/// The current epoch
+		EpochIndex get(epoch_index): u64;
 	}
 }
 
 decl_module! {
-	pub struct Module<T: Trait> for enum Call where origin: T::Origin { }
+	/// The BABE SRML module
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		fn on_initialize() {
+			for i in Self::get_inherent_digests()
+				.logs
+				.iter()
+				.filter_map(|s| s.as_pre_runtime())
+				.filter_map(|(engine, mut data)| if engine == BABE_ENGINE_ID {
+					Decode::decode(&mut data)
+				} else { None }) {
+				let (ref vrf_output, ref _vrf_proof, ref _author, _slot_num): (
+					[u8; VRF_OUTPUT_LENGTH],
+					[u8; VRF_PROOF_LENGTH],
+					[u8; PUBLIC_KEY_LENGTH],
+					u64,
+				) = i;
+				Self::deposit_vrf_output(vrf_output);
+			}
+		}
+	}
+}
+
+impl<T: Trait> RandomnessBeacon for Module<T> {
+	/// The randomness we have right now.
```

Comment:
Having documentation on an implementation is not that useful, IMHO. You already have a similar documentation above the trait function declaration. Maybe you move some stuff from here to there and remove it here.